spec/models/partner_spec.rb

```diff
@@ -17,10 +17,25 @@
   it { is_expected.to belong_to :province }
   it { is_expected.to belong_to :status }
 
-  it { is_expected.to allow_value(Date.current, Date.yesterday).for :start_date }
-  it { is_expected.not_to allow_value(Date.tomorrow).for :start_date }
-  [7, 'yes', true].each do |bad_date_value|
-    it { is_expected.to_not allow_value(bad_date_value).for :start_date }
+  context 'start_date validation' do
+    describe 'valid date' do
+      [Date.current, "10-10-2012"].each do |good_date_value|
+        it { is_expected.to allow_value(good_date_value).for :start_date }
+      end
+
+      [7, 'yes', true].each do |bad_date_value|
+        it { is_expected.to_not allow_value(bad_date_value).for :start_date }
+      end
+    end
```

Comment:
I am working on refactoring this part.

I am already testing good and bad dates when I test the `valid_date?` helper method. So I only need to test the presence of the `date_not_in_future` validator here.

I would like to see here something like:
`it { is_expected.to validate_date_not_in_future_of :start_date }` (same as: ``it { is_expected.to validate_presence_of :start_date }`) but I can't find an rspec method to directly test custom validators.
I'll keep searching for a solution.

Any help is welcome!